### PR TITLE
Yombunker/changing from interface to base class

### PR DESCRIPTION
**Pull request by yombunker**
- Small fixes
- Adding argument validation
- Changing to a base class the sequence generators

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.3'
+        classpath 'com.android.tools.build:gradle:3.0.0-alpha4'
```
Comment (uDevel) [inline build.gradle]: Would that affect users?

**Review by uDevel — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (yombunker) [reply build.gradle]: I thought I had reverted that. I'll send a new update later today, or I don't know if you can modify it

Comment (yombunker) [inline build.gradle]: I thought I had reverted that. I'll send a new update later today, or I don't know if you can modify it

Comment (yombunker) [reply build.gradle]: i don't see it in the modified files

**Pull request merged**